File: 백준/Silver/9375. 패션왕 신해빈/패션왕 신해빈.py
T = int(input())
for _ in range(T):
    N = int(input())
    d = {}
    for _ in range(N):
        name, types = input().split()
        if types in d:
            d[types] = d[types] + [name]
        else:
            d[types] = [name]
    
    answer = 1
    for a in d.values():
        answer *= len(a)+1
    
    print(answer-1)

File: 백준/Silver/9375. 패션왕 신해빈/패션왕 신해빈.java
import java.io.BufferedReader;
import java.io.FileInputStream;
import java.io.InputStreamReader;
import java.util.HashMap;

public class Main {
	public static void main(String[] args) throws Exception {
		
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		
		int T = Integer.parseInt(br.readLine());
		int N;
		HashMap<String, Integer> map;
		for(int tc=0; tc<T; tc++) {
			N = Integer.parseInt(br.readLine());
			map = new HashMap<>();
			for(int i=0; i<N; i++) {
				String arr[] = br.readLine().split(" ");
				if(map.getOrDefault(arr[1], null) == null) map.put(arr[1], 1);
				else map.put(arr[1], map.get(arr[1])+1);
			}
			
			int answer = 1;
			for(Integer i : map.values()) {
				answer *= (i+1);
			}
			System.out.println(answer-1);
		}
		
	}
}